Docs 3379 update encryption articles (#1617)
truenas/documentation: content/SCALE/SCALETutorials/Storage/Pools/EncryptionSCALE.md

You can create an unencrypted dataset on an encrypted pool and you can create an encrypted dataset on an unencrypted pool if you need to protect data with encryption. 

If you add an encrypted dataset under an unencrypted root dataset and then add child datasets under that encrypted dataset it becomes an encrypted non-root parent to any dataset created under it. 
You can let a nested child dataset inherit the encryption settings from the parent or change the settings for the child dataset.

The other datasets created from the unencrypted root dataset can remain unencrypted unless you choose to add it when you create the dataset.

### Encryption Visual Cues

Dataset encryption can be visually confusing in SCALE. 
SCALE uses different lock-type icons to indicate the encryption state of a root, parent, or child dataset in the tree table on the **[Datasets]({{< relref "DatasetsScreensSCALE.md" >}})** screen.
Each icon displays text labels that explain the state of the dataset when you hover the mouse over the icon.

{{< include file="/_includes/EncryptionIconsSCALE.md" type="page" >}}

> If a dataset inherits encryption from either the root or a non-root parent dataset, the locking icons change to a different type and the mouse hover-over label indicates the encryption is **Locked by ancestor** or **Unlocked by ancestor**. 

Reviewer: ```suggestion
If a dataset inherits encryption from either the root or a non-root parent dataset, the locking icons change to a different type, and the mouse hover-over label indicates the encryption is **Locked by ancestor** or **Unlocked by ancestor**. 
```

Each encrypted dataset includes the **ZFS Encryption** widget on the **Datasets** screen when you select the dataset.

The dataset encryption state is unlocked until you lock it using the **Lock** option on the **ZFS Encryption** widget. After locking the dataset, the icon on the tree table changes to the locked version and the **ZFS Encryption** widget displays the **Unlock** option.

### Inherit Encryption

> Datasets *inherit* encryption which means they use the encryption settings of the parent, whether the parent is the root dataset or a non-root parent dataset with child datasets nested under it.

Reviewer: ```suggestion
Datasets *inherit* encryption, which means they use the encryption settings of the parent, whether the parent is the root dataset or a non-root parent dataset with child datasets nested under it.
```

You can change inherited settings for a dataset when you add the dataset, or you can change inherited encryption settings for an existing dataset using the **Edit** option on the **ZFS Encryption** widget.

## Implementing Encryption
Before creating a pool with encryption make sure you want to encrypt all datasets and data stored on the pool. 

{{< hint warning>}}

> You cannot change a pool from an encrypted to a non-encrypted, you can only change the dataset encryption type in the encrypted pool.

Reviewer: ```suggestion
You cannot change a pool from encrypted to non-encrypted. You can only change the dataset encryption type in the encrypted pool.
```

{{< /hint >}}
If your system does not have enough disks to allow you to create a second storage pool, we recommend that you not use encryption at the pool level. 
You can mix encrypted and unencrypted datasets on an unencrypted pool.
{{< hint warning >}}

> All pool-level encryption is key-based encryption. When prompted, download the encryption key, keep it stored in a safe place and where you can back up the file.

Reviewer: ```suggestion
All pool-level encryption is key-based encryption. When prompted, download the encryption key and keep it stored in a safe place where you can back up the file.
```

Read the warning, select **Confirm**, and then click **I UNDERSTAND**.

A second dialog opens where you click **Download Encryption Key** for the pool encryption key. 

![DownloadEncryptionKeyForPool](/images/SCALE/22.12/DownloadEncryptionKeyForPool.png "Download Encryption Key on Pool Manager") 

Click **Done** to close the window. 
Move the encryption key to safe location where you can back up the file.

Click **Save** to create the pool with encryption.

### Adding Encryption to a New Dataset 

To add encryption to a new dataset, go to **Datasets**.

First, select the root or other dataset on the tree table where you want to add a dataset. 
The default dataset selected when you open the **Datasets** screen is the root dataset of the first pool on the tree table list. 
If you have more than one pool and want to create a dataset in a pool other than the default, select the root dataset for that pool or any dataset under the root where you want to add the new dataset. 

Click  **Add Dataset** to open the **Add Dataset** screen. 

To add a dataset, enter a value in **Name**. 

> Next select the type of **Case Sensitivity** and **Share Type** for the dataset.

Reviewer: ```suggestion
Next, select the type of **Case Sensitivity** and **Share Type** for the dataset.
```

To add encyrption to a dataset, select **Inherit** under **Encryption Options** to clear the checkbox. 
This displays the **Encryption** checkbox preselected. 

![AddDatasetEncryptionOptionsInheritCleared](/images/SCALE/22.12/AddDatasetEncryptionOptionsInheritCleared.png "Add Dataset Encryption Options Clear Inherit") 

> Now decide if you want to use the default encryption type key, and if you want to let the system genearate the encryption key. 

Reviewer: ```suggestion
Now decide if you want to use the default encryption type key and if you want to let the system generate the encryption key. 
```

To use key encryption and your own key, clear the **Generate key** checkbox to display the **Key** field. Enter your key in this field.

![AddDatasetEncryptionKeyfromNonEncrypted](/images/SCALE/22.12/AddDatasetEncryptionKeyfromNonEncrypted.png "Add Key Encryption") 

> To change to a passphrase encryption, click the down arrow and select **Passphrase** from the **Encryption Type** dropdown.

Reviewer: ```suggestion
To change to passphrase encryption, click the down arrow and select **Passphrase** from the **Encryption Type** dropdown.
```

Losing encryption keys or passphrases can result in permanent data loss!
{{< /hint >}}

### Changing Dataset Encryption
You cannot add encryption to an existing dataset. 
You can change the encryption type for an already encrypted dataset using the **Edit** option on the **ZFS Encryption** widget for the dataset.

{{< hint danger >}}
Save any change to the encryption key or passphrase, and update your saved passcodes and keys file, and then back up that file. 
{{< /hint >}} 

To change the encryption type, go to **Datasets**:

1. Select the unlocked, encrypted dataset on the tree table, then click **Edit** on the **ZFS Encryption** widget.
   The **Edit Encryption Options** dialog for the selected dataset displays.

   You must unlock a locked encrypted dataset before you can make changes.
   
   If the dataset inherits encryption settings from a parent dataset, to change this, clear the **Inherit encryption properties from parent** checkbox to display the key type encryption setting options.

   ![EditEncryptionOptionsInheritedSettings](/images/SCALE/22.12/EditEncryptionOptionsInheritedSettings.png "Edit Encryption Window - Inherited")

> 2. Change the encryption settings. Key type options are to change the type from **Key** to **Passphrase** or from generated to manually-entered encryption key.

Reviewer: ```suggestion
2. Change the encryption settings. Key type options are to change the type from **Key** to **Passphrase** or from a generated to a manually-entered encryption key.
```

>    After clearing the **Inherits encryption properties from parent** the default settings display with **Encryption Type** set to **Key** and **Genrate Key** pre-selected.

Reviewer: ```suggestion
   After clearing the **Inherits encryption properties from parent** the default settings display with **Encryption Type** set to **Key** and **Generate Key** pre-selected.
```

### Locking a Dataset

> Select the dataset on the tree table then click **Lock** on the **ZFS Encryption** widget to open the **Lock Dataset** dialog with the dataset full path name.

Reviewer: ```suggestion
Select the dataset on the tree table, then click **Lock** on the **ZFS Encryption** widget to open the **Lock Dataset** dialog with the dataset full path name.
```

![LockDatasetDialog](/images/SCALE/22.12/LockDatasetDialog.png "Lock Dataset")

Use the **Force unmount** option only if you are certain no one is currently accessing the dataset. 

> **Force unmount** boots anyone using the dataset, like a share, out of the dataset so you can lock it. 

Reviewer: ```suggestion
**Force unmount** boots anyone using the dataset (e.g. someone accessing a share) so you can lock it. 
```

Click **Confirm** to activate **Lock**, then click **Lock**.

{{< hint info >}}
You *cannot* use locked datasets.
{{< /hint >}}

### Unlocking a Dataset

To unlock a dataset, go to **Datasets** then select the dataset on the tree table. 
Click **Unlock** on the **ZFS Encryption** widget to open the **Unlock Dataset** screen.

![UnlockDatasetScreen](/images/SCALE/22.12/UnlockDatasetScreen.png "Dataset Unlock Screen")

Type the passphrase into **Dataset Passphrase** and click **Save**. 

Select **Unlock Child Encrypted Roots** to unlock all locked child datasets if they use the same passphrase.

> Select **Force** if the path where the dataset mounts exists but is not empty. When this happens the unlock operation fails. Using **Force** allows the system to rename the existing directory and file where the dataset should mount. This prevents the mount operation from failing. 

Reviewer: ```suggestion
Select **Force** if the dataset mount path exists but is not empty. When this happens, the unlock operation fails. Using **Force** allows the system to rename the existing directory and file where the dataset should mount. This prevents the mount operation from failing. 
```
